otwcode/otwarchive
# AO3-2380 Make autocomplete for tags ignore diacritics

# Pull Request Checklist

  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-2380

## Purpose

Changes autocomplete functionality so that diacritics are ignored when searching for autocomplete possibilities (_i.e._ accents are ignored when searching, so "e" matches "é" and vice-versa).

## Testing Instructions

- Create (canonical) tags of various types both with and without accented characters (_e.g._ "àbc" and "apple")
- Verify that autocomplete for either the plain character "a" or the accented one "à" (or any accented variation "ä") will suggest both tags as possibilities

**Note** This functionality is accomplished by storing in redis autocomplete strings stripped of diacritics (in addition to the lower casing that was already happening to make it case insensitive) via `ActiveSupport::Inflector.transliterate(string)` and then similarly stripping diacritics from input when invoking the autocomplete.  You will need to run the rake task `autocomplete:reload_tag_data` in order to get this functionality for existing tags.

## References

https://api.rubyonrails.org/v5.2.5/classes/ActiveSupport/Inflector.html#method-i-transliterate
